### PR TITLE
create code along lab

### PR DESCRIPTION
**Pull request by joshuabamboo**
@jmburges @AnnJohn @SophieDeBenedetto

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,111 @@
-# other-hash-methods-code-along
+# Higher Level Hash Methods
+
+##Objectives
+* Learn some interesting methods we can call on hashes
```
Comment (jmburges) [inline README.md]: Objectives should be in the SWBAT format. So Students Will Be Able To... So a bit mor specific, which methods are they learning?

File: README.md [hunk 2]
```diff
@@ -1 +1,111 @@
...
+```ruby
+family_members = {mom: "Victoria", dad: "Richard", sister: "Zoe"}
+
+family_members.keys
+#  => [:mom, :dad, :sister]
+```
+
+### `.min`
+
+You can use the `.min` method on a hash to return the key/value *pair* that contains that **lowest** value:
```
Comment (jmburges) [inline README.md]: Mention that it returns a `pair`....so what return type can return two different things? Arrays!

File: README.md [hunk 3]
```diff
@@ -1 +1,111 @@
...
+#  => ["Victoria", "Richard", "Zoe"]
+```
+
+We can see that the `.values` method returns an array of the values of the keys in the hash. 
+
+### `.keys`
+
+This method, not surprisingly, returns an array containing all of the keys in the hash that `.keys` has been called on: 
+
+```ruby
```
Comment (jmburges) [inline README.md:29]: For all of these examples, can you make sure to use the same "domain" as your codealong? makes it a bit easier to make the switch. So I'd just change these to groceries

File: README.md [hunk 4]
```diff
@@ -1 +1,111 @@
...
+groceries = {
+  dairy: ["milk", "yogurt", "cheese"],
+  vegetable: ["carrots", "broccoli", "cucumbers"],
+  meat: ["chicken", "steak", "salmon"],
+  grains: ["rice", "pasta"]
+}
+```
+ 
+## Challenge 1: Return the values
+Use the `.values` method to collect all of the values of the grocery type keys (`:dairy`, `:vegetables`, `:meat`, `:grains`). The method should return a *one-dimensional* (or "flat") array that *only* includes the values (groceries such as "milk" and "carrots") without their keys.
```
Comment (jmburges) [inline README.md:66]: Don't immediately jump into a nested hash. Make the first challenge without the nesting, then jump into nesting

**Comment by jmburges** [comment]
Looks good, try and stay to one domain and warm them up a bit with non nested things first.

Also these need tests! Gotta get those green lights baby.

File: README.md (hunk 4 above)
Comment (joshuabamboo) [inline README.md:66]: I would argue that we should keep this bc it's right in the middle of the nested hash section.

Comment (jmburges) [inline README.md:66]: fine with that. disregard

**Comment by jmburges** [comment]
looks good :shipit:

**Pull request merged**